File: csharp/prime-factors/PrimeFactors.cs
using System;
using System.Collections.Generic;
public class PrimeFactors
{
  public static List<int> For(long n)
  {
    List<int> primes = new List<int>();
    int i = 2;
    while (n > 1)
    {
      while (n % i == 0)
      {
        primes.Add(i);
        n /= i;
      }
      i++;
    }
    return primes;
  }
}

File: ruby/prime-factors/prime_factors.rb
class PrimeFactors
  def self.for(n)
    primes = []
    i = 2
    while n > 1 do
      while n % i == 0 do
        primes << i
        n /= i
      end
      i += 1
    end
    primes
  end
end
